src/core/externalstorage/qgsexternalstorageregistry.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+  qgsexternalstorageregistry.h
+  --------------------------------------
+  Date                 : March 2021
+  Copyright            : (C) 2021 by Julien Cabieces
+  Email                : julien dot cabieces at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSEXTERNALSTORAGEREGISTRY_H
+#define QGSEXTERNALSTORAGEREGISTRY_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+
+#include <QList>
+#include <QHash>
+
+class QgsExternalStorage;
+
+/**
+ * \ingroup core
+ * \brief Registry of external storage backends used by QgsExternalResourceWidget
+ *
+ * This is a singleton that should be accessed through QgsApplication::externalStorageRegistry().
```

Comment:
Sorry but I think I missed the singleton implementation. Do we have a `instance()` method somehwere?

Reply:
This is not strictly a singleton because the constructor is not private and the instance is created in QgsApplication. I changed the comment.